src/application/router/index.ts

```diff
@@ -1,10 +1,18 @@
-import { createRouter, createWebHistory } from 'vue-router';
-import routes from './routes';
+import { createRouter, createWebHistory, Router } from 'vue-router';
+import createRoutes from './routes';
+import NoteService from '../../domain/note.service';
 
-const router = createRouter({
-  history: createWebHistory(),
-  routes,
-});
-
-export default router;
+/**
+ * Init router
+ *
+ * @param noteService - Note service instance
+ * @returns { Router } - Router
+ */
+export default function (noteService: NoteService): Router {
```

Comment:
I think the passing of Domain Services from the Router through the Components to the Application Services is a bad flow. 

Components shouldn't know about Domain Services.

Let's try moving the initialization of services to the `domain/index.ts`?

Reply:
Can i use dependcy inject for note service to use it in `useNote`? It it a good practice?

Reply:
I think you can simply import them

Reply:
done